pkg/internal/controller/controller.go

```diff
@@ -34,13 +34,18 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	"sigs.k8s.io/controller-runtime/pkg/source"
+	"sigs.k8s.io/logical-cluster"
 )
 
 // Controller implements controller.Controller.
 type Controller struct {
 	// Name is used to uniquely identify a Controller in tracing, logging and monitoring.  Name is required.
 	Name string
 
+	// Cluster is the logical cluster that this controller is running against.
+	// +optional
+	Cluster logical.Name
```

Comment:
This might be a stepping stone, and I should probably have added more information. It's fair to say that we probably want a Controller to receive requests from many clusters all at once; and the Request to scope it.

That said, right now given the complex flow of operations, I opted for each Controller to be scoped to a single Cluster, that way we can validate that events have all a Cluster field populated (or panic). I'm planning to remove this at some point, just need to think through it a bit more. If you have any suggestions feel free to ping 😄